### PR TITLE
fix: only generate typing files with sync command

### LINKED ISSUE #8525: npm run check make app reload and CSP blocks the navigation
### Describe the bug

I start development. The app starts in the browser. Everything works.

I launch in another shell's console "`npm run check`" and then the browser reloads the page and I cannot use more the app because of the below errors (I'm using CSP):

```shell
Refused to connect to 'ws://localhost:4000/' because it violates the following Content Security Policy directive: "default-src https://*.project.com". Note that 'connect-src' was not explicitly set, so 'default-src' is used as a fallback.
```

In my `svelte.config.js` I have this:

```js
import adapter from "@sveltejs/adapter-static";
import { vitePreprocess } from "@sveltejs/kit/vite";
import "dotenv/config";

const dev = process.env.NODE_ENV === "development";

/** @type {import('@sveltejs/kit').Config} */
const config = {
  preprocess: vitePreprocess(),

  kit: {
    adapter: adapter({
      fallback: "index.html",
    }),

    serviceWorker: {
      register: !dev,
    },

    csp: dev
      ? undefined
      : {
          directives: {
            "default-src": [`https://*.${process.env.PUBLIC_PROJECT_URL}`],
            "style-src": [
              `https://*.${process.env.PUBLIC_PROJECT_URL}`,
              "unsafe-inline",
            ],
            "img-src": ["*", "data:"],
          },
        },
  },
};

export default config;
```

I think `npm run check` is restarting the app without `process.env.NODE_ENV === 'development'`.

My `npm run check` is: `"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json"`.

**Is there a way to disable CSP during development?**

### System Info

```Shell
System:
    OS: Windows 10 10.0.19045
  Binaries:
    Node: 18.13.0
    npm: 8.19.3
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (107.0.1418.62)
  npmPackages:
    @sveltejs/adapter-static: 1.0.2 => 1.0.2
    @sveltejs/kit: 1.1.0 => 1.1.0
    svelte: 3.55.1 => 3.55.1
    vite: 4.0.4 => 4.0.4
```

### Severity

blocking an upgrade

### Additional Information

This didn't happen with `@sveltejs/kit@1.0.2`.

### PR DESCRIPTION
**Pull request by dummdidumm**
fixes #8525

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: .changeset/neat-geckos-end.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: only generate typing files with sync command
```
Comment (benmccann) [inline .changeset/neat-geckos-end.md:5]:
```suggestion
fix: only generate type definitions with `sync` command
```

File: packages/kit/src/cli.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const prog = sade('svelte-kit').version(pkg.version);
 
 prog
 	.command('sync')
-	.describe('Synchronise generated files')
+	.describe('Synchronise generated typing files')
```
Comment (benmccann) [inline packages/kit/src/cli.js:27]:
```suggestion
	.describe('Synchronise generated type definitions')
```

**Comment by benmccann** [comment]
https://kit.svelte.dev/docs/cli#svelte-kit-sync probably needs to be updated as well

**Comment by benmccann** [comment]
I wonder if this will stop `$app` from being generated causing issues in VS Code?

**Comment by dummdidumm** [comment]
Vs code needs the type definition of `$app`, which is available through this, so no.

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**